opentelemetry-sdk/tests/resources/test_resources.py

```diff
@@ -158,21 +147,56 @@ def test_aggregated_resources(self):
             ),
         )
 
+    def test_resource_detector_ignore_error(self):
         resource_detector = mock.Mock(spec=resources.ResourceDetector)
         resource_detector.detect.side_effect = Exception()
         resource_detector.raise_on_error = False
         self.assertEqual(
-            resources.get_aggregated_resources(
-                [resource_detector, resource_detector1]
-            ),
-            resources.Resource({"key1": "value1"}),
+            resources.get_aggregated_resources([resource_detector]),
+            resources.Resource.create_empty(),
         )
 
+    def test_resource_detector_raise_error(self):
         resource_detector = mock.Mock(spec=resources.ResourceDetector)
         resource_detector.detect.side_effect = Exception()
         resource_detector.raise_on_error = True
         self.assertRaises(
-            Exception,
-            resources.get_aggregated_resources,
-            [resource_detector, resource_detector1],
+            Exception, resources.get_aggregated_resources, [resource_detector],
+        )
+
+
+class TestOTELResourceDetector(unittest.TestCase):
+    def setUp(self) -> None:
+        os.environ["OTEL_RESOURCE"] = ""
+
+    def tearDown(self) -> None:
+        os.environ.pop("OTEL_RESOURCE")
+
+    def test_otel_resource_detector_empty(self):
```

Comment:
Now that this is in a class called "TestOTELResourceDetector", keeping "test_otel_resource_detector_" in the test names seems redundant. We can probably shorten these names.